postgres/server/init.sls

```diff
@@ -86,23 +86,43 @@ postgresql-config-dir:
     - require:
       - cmd: postgresql-cluster-prepared
 
-{%- if postgres.postgresconf %}
+{%- set db_port = salt['config.option']('postgres.port') %}
+{%- if db_port %}
+
+postgresql-conf-comment-port:
+  file.comment:
+    - name: {{ postgres.conf_dir }}/postgresql.conf
+    - regex: ^port\s*=.+
+    - require:
+      - file: postgresql-config-dir
+
+{%- endif %}
+
+{%- if postgres.postgresconf or db_port %}
 
 postgresql-conf:
   file.blockreplace:
     - name: {{ postgres.conf_dir }}/postgresql.conf
     - marker_start: "# Managed by SaltStack: listen_addresses: please do not edit"
```

Comment:
Obviously can't change this marker to mention that the managed block is also for the `port`.